pkg/scheduler/internal/cache/interface.go

```diff
@@ -57,7 +57,8 @@ import (
 // - Both "Expired" and "Deleted" are valid end states. In case of some problems, e.g. network issue,
 //   a pod might have changed its state (e.g. added and deleted) without delivering notification to the cache.
 type Cache interface {
-	// ListPods lists all pods in the cache.
+	// ListPods returns all the pods in the cache (including those from deleted nodes).
+	// DO NOT use outside of tests.
```

Comment:
It seems `TestSchedulerNoPhantomPodAfterExpire` in pkg/scheduler/scheduler_test.go is the only consumer of `ListPods()`. And that test only verifies if len(pods) == 0, so ideally, we can remove the signature of ListPods() by either (1) refactoring the test case, or (2) providing a function like `PodsLen()`. WDYT?

Reply:
Went with 2.